extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/utils.js

```diff
@@ -0,0 +1,463 @@
+  getLocalizedString: function(language, stringId, localizedStrings) {
+    try {
+      return localizedStrings[language][stringId];
+    } catch (e) {
+      return "";
+    }
+
+  },
+
+  /**
+   * Safely gets the value of an object's nested property.
+   *
+   * @function getPropertyValue
+   * @param {Object} object - The object we want to extract the property form
+   * @param {String} propertyPath - A path that points to a nested property in the object with a form like 'prop.nestedProp1.nestedProp2'
+   * @param {Object} defaltValue - (Optional) A default value to return when the property is undefined
+   * @return {Object} - The value of the nested property, the default value if nested property was undefined
+   */
+  getPropertyValue: function(object, propertyPath, defaltValue) {
+    var value = null;
+    var currentObject = object;
+    var currentProp = null;
+
+    try {
+      var props = propertyPath.split('.');
+
+      for (var i = 0; i < props.length; i++) {
+        currentProp = props[i];
+        currentObject = value = currentObject[currentProp];
+      }
+      return value || defaltValue;
+    } catch (err) {
+      return defaltValue;
+    }
+  },
+
+  /**
+  * Highlight the given element for hover effects
+  *
+  * @function Highlight
+  * @param {DOMElement} target - The element to Highlight
+  */
+  highlight: function(target, opacity, color) {
+    target.style.opacity = opacity;
+    target.style.color = color;
+    target.style.WebkitFilter = "drop-shadow(0px 0px 3px rgba(255,255,255,0.8))";
+    target.style.filter = "drop-shadow(0px 0px 3px rgba(255,255,255,0.8))";
+    target.style.msFilter = "progid:DXImageTransform.Microsoft.Dropshadow(OffX=0, OffY=0, Color='#fff')";
+  },
+
+  /**
+  * Remove the highlight effect of the given element
+  *
+  * @function removeHighlight
+  * @param {DOMElement} target - The element to remove the highlight effect from
+  * @param {DOMElement} opacity - The opacity to return the element to
+  */
+  removeHighlight: function(target, opacity, color) {
+    target.style.opacity = opacity;
+    target.style.color = color;
+    target.style.WebkitFilter = "";
+    target.style.filter = "";
+    target.style.msFilter = "";
+  },
+
+  /**
+  * Determine which buttons should be shown in the control bar given the width of the player<br/>
+  * Note: items which do not meet the item spec will be removed and not appear in the results.
+  *
+  * @function collapse
+  * @param {Number} barWidth - Width of the control bar
+  * @param {Object[]} orderedItems - array of left to right ordered items. Each item meets the skin's "button" schema.
+  * @returns {Object} An object of the structure {fit:[], overflow:[]} where the fit object is
+  *   an array of buttons that fit in the control bar and overflow are the ones that should be hidden
+  */
+  collapse: function( barWidth, orderedItems, responsiveUIMultiple ) {
+    if( isNaN( barWidth ) || barWidth === undefined ) { return orderedItems; }
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this "===" check; it will always be false. Did you mean to use "=="? [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS3403)